lib/from_erlang.ml

```diff
@@ -271,21 +283,21 @@ and function_of_clauses clauses =
     in
     let (cs, arities) = clauses |> List.map ~f:(function
     | F.ClsCase _ -> failwith "cannot reach here"
-    | F.ClsFun {patterns; body; _} ->
+    | F.ClsFun {line; patterns; body; _} ->
       (* Ignore guards currently since guard is complex and it's not needed for simple examples *)
       let ps = patterns |> List.map ~f:pattern_of_erlang_pattern in
       let arity = List.length ps in
       let tuple_pattern = PatTuple ps in
-      (((tuple_pattern, Constant (Atom ("true"))), expr_of_erlang_expr' body), arity)
+      (((tuple_pattern, Constant (line, Atom ("true"))), expr_of_erlang_expr' body), arity)
     ) |> List.unzip in
      let make_fresh_variables length = fill (fun () -> Variable.create()) length |> List.rev in
      let make_case cs fresh_variables =
-       let fresh_tuple = Tuple (fresh_variables |> List.map ~f:(fun v -> Var v)) in
+       let fresh_tuple = Tuple (fresh_variables |> List.map ~f:(fun v -> Var (-1, v))) in
        (* letrec $name = fun $name(A1, A2, ...) -> b1; $name(B1, B2, ...)-> b2; ... end in $name *)
        Case (fresh_tuple, cs)
      in
      match cs with
-     | ((PatTuple patterns, Constant (Atom ("true"))), body)::[] ->
+     | ((PatTuple patterns, Constant (-1, Atom ("true"))), body)::[] ->
```

Comment:
I suspect that this pattern is never matched because `cs` contains `Constant (line, Atom ("true"))` whose `line` is from `ClsFun` but this pattern's line number is always `-1`.

I suggest:

```suggestion
     | ((PatTuple patterns, Constant (_line, Atom ("true"))), body)::[] ->
```